File: CompanionLib/Utility/FBXCTestReporterConfiguration.m
/*
 * Copyright (c) Meta Platforms, Inc. and affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

#import "FBXCTestReporterConfiguration.h"

#import <FBControlCore/FBControlCore.h>
#import <XCTestBootstrap/XCTestBootstrap.h>

@implementation FBXCTestReporterConfiguration

- (instancetype)initWithResultBundlePath:(nullable NSString *)resultBundlePath coverageConfiguration:(nullable FBCodeCoverageConfiguration *)coverageConfiguration logDirectoryPath:(nullable NSString *)logDirectoryPath binariesPaths:(nullable NSArray<NSString *> *)binariesPaths reportAttachments:(BOOL)reportAttachments reportResultBundle:(BOOL)reportResultBundle
{
  self = [super init];
  if (!self) {
    return nil;
  }

  _resultBundlePath = resultBundlePath;
  _coverageConfiguration = coverageConfiguration;
  _logDirectoryPath = logDirectoryPath;
  _binariesPaths = binariesPaths;
  _reportAttachments = reportAttachments;
  _reportResultBundle = reportResultBundle;

  return self;
}

- (NSString *)description
{
  return [NSString stringWithFormat:@"Result Bundle %@ | Coverage %@ | Log Path %@ | Binaries Paths %@ | Report Attachments %d | Report Restul Bundle %d", self.resultBundlePath, self.coverageConfiguration, self.logDirectoryPath, [FBCollectionInformation oneLineDescriptionFromArray:self.binariesPaths], self.reportAttachments, self.reportResultBundle];
}

@end

File: CompanionLib/Utility/FBXCTestReporterConfiguration.swift
/*
 * Copyright (c) Meta Platforms, Inc. and affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

import FBControlCore
import Foundation
import XCTestBootstrap

public struct FBXCTestReporterConfiguration {

  public let resultBundlePath: String?
  public let coverageConfiguration: FBCodeCoverageConfiguration?
  public let logDirectoryPath: String?
  public let binariesPaths: [String]
  public let reportAttachments: Bool
  public let reportResultBundle: Bool

  public init(resultBundlePath: String?, coverageConfiguration: FBCodeCoverageConfiguration?, logDirectoryPath: String?, binariesPaths: [String]?, reportAttachments: Bool, reportResultBundle: Bool) {
    self.resultBundlePath = resultBundlePath
    self.coverageConfiguration = coverageConfiguration
    self.logDirectoryPath = logDirectoryPath
    self.binariesPaths = binariesPaths ?? []
    self.reportAttachments = reportAttachments
    self.reportResultBundle = reportResultBundle
  }

}

// MARK: - CustomStringConvertible

extension FBXCTestReporterConfiguration: CustomStringConvertible {

  public var description: String {
    let coverageDesc = coverageConfiguration.map { "\($0)" } ?? "(null)"
    return "Result Bundle \(resultBundlePath ?? "(null)") | Coverage \(coverageDesc) | Log Path \(logDirectoryPath ?? "(null)") | Binaries Paths \(FBCollectionInformation.oneLineDescription(from: binariesPaths)) | Report Attachments \(reportAttachments ? 1 : 0) | Report Restul Bundle \(reportResultBundle ? 1 : 0)"
  }
}
